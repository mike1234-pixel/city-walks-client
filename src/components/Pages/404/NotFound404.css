.nf404-page-container {
  width: 100%;
  text-align: center;
}

/*# sourceMappingURL=NotFound404.css.map */
