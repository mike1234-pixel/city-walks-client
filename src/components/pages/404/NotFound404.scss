.nf404-page-container {
  width: 100%;
  text-align: center;
}
