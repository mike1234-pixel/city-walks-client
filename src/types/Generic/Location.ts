export default interface Location {
    hash: string;
    pathname: string;
    search: string;
    state: undefined;
}