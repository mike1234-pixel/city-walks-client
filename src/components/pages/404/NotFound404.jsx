import './NotFound404.css'

const nf404 = ({ location }) => {
    const errorMessage = `404 Error: Page not found at ${location.pathname}`
    return (
        <div className="nf404-page-container min-page-height">
            <div className="page-heading-container">
                <h1 className="page-heading">{errorMessage}</h1>
            </div>
        </div>
    )
}

export default nf404