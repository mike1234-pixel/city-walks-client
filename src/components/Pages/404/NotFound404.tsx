import { MDBContainer } from "mdbreact";
import Location from "../../../types/Generic/Location";
import "./NotFound404.css";

interface nf404Props {
  location: Location;
}

const nf404 = (props: nf404Props) => {

  const { location } = props;

  const errorMessage = `404 Error: Page not found at ${location.pathname}`;

  return (
    <MDBContainer>
      <div className='nf404-page-container page'>
        <div className='page-heading-container'>
          <h1 className='page-heading'>{errorMessage}</h1>
        </div>
      </div>
    </MDBContainer>
  );
};

export default nf404;
