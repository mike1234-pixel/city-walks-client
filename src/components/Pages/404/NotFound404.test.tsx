import renderer, { ReactTestRendererJSON } from 'react-test-renderer'
import Location from "../../../types/Generic/Location"
import NotFound404 from './NotFound404'

// snapshot test
it('NotFound404 Page renders correctly', () => {

  const location: Location = {
    hash: '',
    key: '',
    pathname: '/pagedoesntexist',
    search: '',
    state: null
  }

  const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
    .create(<NotFound404 location={location} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});