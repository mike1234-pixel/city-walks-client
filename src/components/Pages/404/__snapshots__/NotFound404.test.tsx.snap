// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound404 Page renders correctly 1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="nf404-page-container page"
  >
    <div
      className="page-heading-container"
    >
      <h1
        className="page-heading"
      >
        404 Error: Page not found at /pagedoesntexist
      </h1>
    </div>
  </div>
</div>
`;
